### PR TITLE
Inbox list for teams desktop

### PR DESCRIPTION
**Pull request by chrisnojima**
This is a first pass at the better inbox list for teams. There's 2 modes (expanded or not) which shows all the simple teams or a truncated list.
If the big teams are off screen there's a button on the bottom you can click to collapse the simple teams
The icon in the expanded state is the keybase icon but thats temp until the real asset exist
(cc: @cecileboucheron  we don't have the up arrow in the iconfont)

I also added glamorous to ./styles so we can pull it in correctly from native/desktop

There's some TODO placeholder stuff for mobile. As-is it works like it did before but features of this pr aren't there in this PR (except the filtering logic)

### CODE REVIEW

File: shared/chat/inbox/container.js [hunk 1]
```diff
@@ -56,9 +57,10 @@ const filteredInbox = createImmutableEqualSelector(
     // convert any small teams into smallids
     Object.keys(bigTeamToChannels).forEach(team => {
       // only one channel
-      if (Object.keys(bigTeamToChannels[team]) === 1) {
-        smallIds.push(bigTeamToChannels[team])
-        bigTeamToChannels[team] = undefined
+      const channels = Object.keys(bigTeamToChannels[team])
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: this wasn't working before

File: shared/chat/inbox/container.js [hunk 2]
```diff
@@ -78,36 +80,67 @@ const filteredInbox = createImmutableEqualSelector(
...
-  isLoading: state.chat.get('inboxUntrustedState') === 'loading',
-  rows: getRows(state),
-  showNewConversation: state.chat.inSearch && state.chat.inboxSearch.isEmpty(),
-})
+const mapStateToProps = (state: TypedState, {isActiveRoute, smallTeamsExpanded}) => {
+  const [allSmallTeams, bigTeams] = getRows(state, smallTeamsExpanded)
+
+  let smallTeams = allSmallTeams
+  let showSmallTeamsExpandDivider = false
+  if (bigTeams.count() && allSmallTeams.count() > smallteamsCollapsedMaxShown) {
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: figure out if we should show the divider (don't show if there are no big teams or if your small teams list isn't big enough

File: shared/chat/inbox/container.js [hunk 3]
```diff
@@ -78,36 +80,67 @@ const filteredInbox = createImmutableEqualSelector(
...
+    showSmallTeamsExpandDivider = true
+    if (!smallTeamsExpanded) {
+      smallTeams = allSmallTeams.slice(0, smallteamsCollapsedMaxShown)
+    }
+  }
+
+  const bigTeamsBadgeCount = 0 // TODO real number
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: do this later

File: shared/chat/inbox/index.desktop.js [hunk 4]
```diff
@@ -56,46 +57,113 @@ class NewConversation extends PureComponent<void, {}, void> {
...
         isActiveRoute={true}
         teamname={row.teamname}
         channelname={row.channelname}
+        type={row.type}
       />
     )
   }
 
-  _onScroll = debounce(() => {
+  _updateShowFloating = () => {
```
Comment (chrisnojima) [inline shared/chat/inbox/index.desktop.js]: how we figure out whats visible

File: shared/chat/inbox/index.desktop.js [hunk 5]
```diff
@@ -115,38 +183,51 @@ class Inbox extends PureComponent<void, Props, void> {
...
             itemSizeGetter={this._itemSizeGetter}
           />
         </div>
+        {this.state.showFloating &&
+          this.props.showSmallTeamsExpandDivider &&
+          <FloatingDivider
+            toggle={this.props.toggleSmallTeamsExpanded}
+            badgeCount={this.props.bigTeamsBadgeCount}
+          />}
+        {/*
```
Comment (chrisnojima) [inline shared/chat/inbox/index.desktop.js]: this is in the design but it should lead to the teams tab which doesn't exist yet

File: shared/chat/inbox/row/container.js [hunk 6]
```diff
@@ -145,8 +145,8 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => ({
 const ConnectedRow = compose(
   // $FlowIssue
   pausableConnect(mapStateToProps, mapDispatchToProps, mergeProps),
-  branch(props => props.teamname && !props.channelname, renderComponent(TeamRow)),
-  branch(props => props.teamname && props.channelname, renderComponent(ChannelRow))
+  branch(props => props.type === 'bigHeader', renderComponent(TeamRow)),
```
Comment (chrisnojima) [inline shared/chat/inbox/row/container.js]: we have an explicit type now since the list itself makes some decisions the row doesn't know

**Comment by keybase-ci-visdiff** [comment]
The commits 282d04896af7...4b003f787284 introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Chat_Inbox-You_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503420720661-282d04896af7-4b003f787284-linux/Chat_Inbox-You_Rekey_Selected-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503420720661-282d04896af7-4b003f787284-linux/Chat_Inbox-You_Rekey_Selected-4b003f787284.png)  
   Chat_Inbox-You_Rekey_Selected comparison
 * Changed: **Chat_Inbox-You_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503420720661-282d04896af7-4b003f787284-linux/Chat_Inbox-You_Rekey-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503420720661-282d04896af7-4b003f787284-linux/Chat_Inbox-You_Rekey-4b003f787284.png)  
   Chat_Inbox-You_Rekey comparison
</summary>

File: shared/chat/inbox/container.js [hunk 7]
```diff
@@ -19,30 +20,34 @@ const getFilter = (state: TypedState) => state.chat.get('inboxFilter')
 
 const createImmutableEqualSelector = createSelectorCreator(defaultMemoize, I.is)
 
-const passesFilter = (i: Constants.InboxState, filter: string): boolean => {
+const passesStringFilter = (filter: string, toCheck: string): boolean => {
+  return toCheck.toLowerCase().indexOf(filter.toLowerCase()) >= 0
+}
+
+const passesParticipantFilter = (participants: I.List<string>, filter: string, you: ?string): boolean => {
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: we need to filter ourselves out as we don't appear in the list of participants and its weird for that to match

File: shared/chat/inbox/container.js [hunk 8]
```diff
@@ -56,12 +61,35 @@ const filteredInbox = createImmutableEqualSelector(
...
+          smallIds.push(bigTeamToChannels[team][channels[0]])
+        }
+        delete bigTeamToChannels[team]
       }
     })
 
+    // Filter out big teams
+    if (filter) {
+      Object.keys(bigTeamToChannels).forEach(team => {
+        // teamname doesn't pass
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: if you type a teamname, show all its channels

File: shared/chat/inbox/container.js [hunk 9]
```diff
@@ -75,39 +103,78 @@ const filteredInbox = createImmutableEqualSelector(
...
+
+    const pids = I.List(pending.keySeq().map(k => ({conversationIDKey: k, type: 'small'})))
+    const sids = I.List(smallIds.map(s => ({conversationIDKey: s, type: 'small'})))
+
+    const bigTeams = I.List(
+      flatten(
+        Object.keys(bigTeamToChannels).sort().map(team => {
+          const channels = bigTeamToChannels[team]
+          return [
+            ...(filter
```
Comment (chrisnojima) [inline shared/chat/inbox/container.js]: no header if we're filtered

File: shared/chat/inbox/index.desktop.js [hunk 10]
```diff
@@ -56,47 +57,119 @@ class NewConversation extends PureComponent<{}> {
...
     }
   }
 
   _itemSizeGetter = index => {
     const row = this.props.rows.get(index)
-    if (row.teamname) {
-      return 24
-    } else {
-      return 56
+    switch (row.type) {
```
Comment (chrisnojima) [inline shared/chat/inbox/index.desktop.js]: we have different heights now

File: shared/chat/inbox/row/big-team-rows.js [hunk 11]
```diff
@@ -3,12 +3,13 @@ import React, {PureComponent} from 'react'
...
 import {globalStyles, globalColors, globalMargins} from '../../../styles'
 import {isMobile} from '../../../constants/platform'
+import {TeamAvatar} from './avatars'
 
 type TeamProps = {
   teamname: string,
 }
 
-class TeamRow extends PureComponent<TeamProps> {
+class BigTeamHeaderRow extends PureComponent<TeamProps> {
```
Comment (chrisnojima) [inline shared/chat/inbox/row/big-team-rows.js]: some small renaming

File: shared/chat/inbox/row/small-team-rows.js [hunk 12]
```diff
@@ -19,25 +19,28 @@ type SimpleProps = {
...
         <Box style={{...rowContainerStyle, backgroundColor: props.backgroundColor}}>
-          <Avatars
-            backgroundColor={props.backgroundColor}
-            isMuted={props.isMuted}
-            isSelected={props.isSelected}
-            participantNeedToRekey={props.participantNeedToRekey}
-            participants={props.participants}
-            youNeedToRekey={props.youNeedToRekey}
-          />
+          {props.teamname
```
Comment (chrisnojima) [inline shared/chat/inbox/row/small-team-rows.js]: the simple team and adhoc teams basically look the same but very slightly different, just keeping this in one place

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready. i'm starting to mobile side as a branch off of this now

**Comment by keybase-ci-visdiff** [comment]
The commits 282d04896af7...4ca38e43e965 introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Chat_Inbox-You_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503431678471-282d04896af7-4ca38e43e965-linux/Chat_Inbox-You_Rekey_Selected-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503431678471-282d04896af7-4ca38e43e965-linux/Chat_Inbox-You_Rekey_Selected-4ca38e43e965.png)  
   Chat_Inbox-You_Rekey_Selected comparison
 * Changed: **Chat_Inbox-You_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503431678471-282d04896af7-4ca38e43e965-linux/Chat_Inbox-You_Rekey-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503431678471-282d04896af7-4ca38e43e965-linux/Chat_Inbox-You_Rekey-4ca38e43e965.png)  
   Chat_Inbox-You_Rekey comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 282d04896af7...44ecf0565862 introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Chat_Inbox-You_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503498927982-282d04896af7-44ecf0565862-linux/Chat_Inbox-You_Rekey_Selected-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503498927982-282d04896af7-44ecf0565862-linux/Chat_Inbox-You_Rekey_Selected-44ecf0565862.png)  
   Chat_Inbox-You_Rekey_Selected comparison
 * Changed: **Chat_Inbox-You_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503498927982-282d04896af7-44ecf0565862-linux/Chat_Inbox-You_Rekey-282d04896af7.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1503498927982-282d04896af7-44ecf0565862-linux/Chat_Inbox-You_Rekey-44ecf0565862.png)  
   Chat_Inbox-You_Rekey comparison
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump open for 2 days

File: shared/chat/inbox/index.desktop.js [hunk 13]
```diff
@@ -56,47 +57,119 @@ class NewConversation extends PureComponent<{}> {
...
-  _onScroll = debounce(() => {
+  _updateShowFloating = () => {
+    let showFloating = true
+    if (this._list) {
+      const [, last] = this._list.getVisibleRange()
+      if (typeof last === 'number') {
+        const row = this.props.rows.get(last)
+
+        if (!row || row.type !== 'small') {
+          showFloating = false
```
Comment (MarcoPolo) [inline shared/chat/inbox/index.desktop.js]: screen shot 2017-08-24 at 12 59 42 pm

File: shared/chat/inbox/row/big-team-rows.js [hunk 14]
```diff
@@ -71,6 +72,60 @@ class ChannelRow extends PureComponent<ChannelProps> {
   }
 }
 
+type FilteredChannelProps = {
+  teamname: string,
+  channelname: string,
+  onSelectConversation: () => void,
+}
+
+class BigTeamChannelFilteredRow extends PureComponent<FilteredChannelProps> {
```
Comment (MarcoPolo) [inline shared/chat/inbox/row/big-team-rows.js]: Why a class?

**Comment by MarcoPolo** [comment]
I also would like to see a story that looks more like what you see for a real user (with several personal convs and several channels and teams) so you can see the floating tab stuff

File: shared/chat/inbox/index.desktop.js (hunk 13 above)
Comment (chrisnojima) [inline shared/chat/inbox/index.desktop.js]: Well, its supposed to stick if there are teams below it. Could be cause of lag?

**Pull request merged**